pkg/network/tracer/testutil/testdns/test_dns_server.go

```diff
@@ -10,30 +10,33 @@ package testdns
 
 import (
 	"net"
+	"strconv"
 	"sync"
 	"testing"
 
 	"github.com/miekg/dns"
 	"github.com/stretchr/testify/require"
 )
 
-var globalServer *server
-var globalServerError error
 var serverOnce sync.Once
 
+const localhostAddr = "127.0.0.1"
+
 // GetServerIP returns the IP address of the test DNS server. The test DNS server returns canned responses for several
 // known domains that are used in integration tests.
 //
 // see server#start to see which domains are handled.
 func GetServerIP(t *testing.T) net.IP {
+	var err error
```

Comment:
What will happen if:

* test #1 calls GetServerIP and the sync.Once function fails 
* test #2 calls GetServerIP
?

Reply:
I see what you mean. I think we need to keep the globalError and have it set by the call to `ListenAndServe()`